hikari/impl/special_endpoints.py

```diff
@@ -944,13 +977,39 @@ def set_default_permission(self: _CommandBuilderT, state: undefined.UndefinedOr[
     def build(self, entity_factory: entity_factory_.EntityFactory, /) -> data_binding.JSONObject:
         data = data_binding.JSONObjectBuilder()
         data["name"] = self._name
-        data["description"] = self._description
+        data.put("description", self._description)
+        data.put("type", self._type)
```

Comment:
I didn't notice this before but this current approach seems a bit odd, why not  move `description` and `options` to `SlashCommandBuilder` and have `type` be a static type that's decided by which builder you're using rather than passed through (where you'd do something like 
```py
data = super().build(entity_factory)
data["type"] = commands.CommandType.SLASH_COMMAND
...
return data
```